src/Registry/Registry.jl

```diff
@@ -293,17 +298,16 @@ function update(regs::Vector{RegistrySpec} = RegistrySpec[]; io::IO=DEFAULT_IO[]
                 if url !== nothing && (new_hash = pkg_server_url_hash(url)) != old_hash
                     let new_hash = new_hash
                         # TODO: update faster by using a diff, if available
-                        mktempdir() do tmp
-                            try
-                                download_verify_unpack(url, nothing, tmp, ignore_existence = true)
-                            catch err
-                                @error "could not download $url" exception=err
-                            end
-                            tree_info_file = joinpath(tmp, ".tree_info.toml")
-                            hash = pkg_server_url_hash(url)
-                            write(tree_info_file, "git-tree-sha1 = " * repr(string(new_hash)))
-                            cp(tmp, reg.path, force=true)
+                        tmp = mktempdir(cleanup=false) # no cleanup since we mv it
```

Comment:
Cleanup shouldn't do anything if the temp path is already gone, so it should be safe to leave cleanup on in case it somehow doesn't get cleaned up otherwise.

Reply:
Ok, makes sense. Updated